src/metis/error.c

```diff
@@ -125,69 +125,71 @@ int gk_siguntrap()
 
   return 1;
 }
-  
+
 
 /*************************************************************************/
 /*! This function is the custome signal handler, which all it does is to
-    perform a longjump to the most recent saved environment 
+    perform a longjump to the most recent saved environment
  */
 /*************************************************************************/
 void gk_sigthrow(int signum)
 {
   longjmp(gk_jbufs[gk_cur_jbufs], signum);
 }
-  
+
 
 /***************************************************************************
-* This function sets a number of signal handlers and sets the return point 
+* This function sets a number of signal handlers and sets the return point
 * of a longjmp
 ****************************************************************************/
-void gk_SetSignalHandlers() 
+void gk_SetSignalHandlers()
 {
   old_SIGMEM_handler = signal(SIGMEM,  gk_NonLocalExit_Handler);
   old_SIGERR_handler = signal(SIGERR,  gk_NonLocalExit_Handler);
 }
-  
+
 
 /***************************************************************************
 * This function sets the handlers for the signals to their default handlers
 ****************************************************************************/
-void gk_UnsetSignalHandlers() 
+void gk_UnsetSignalHandlers()
 {
   signal(SIGMEM,  old_SIGMEM_handler);
   signal(SIGERR,  old_SIGERR_handler);
 }
-  
+
 
 /*************************************************************************
-* This function is the handler for SIGUSR1 that implements the cleaning up 
+* This function is the handler for SIGUSR1 that implements the cleaning up
 * process prior to a non-local exit.
 **************************************************************************/
 void gk_NonLocalExit_Handler(int signum)
 {
   longjmp(gk_jbuf, signum);
 }
-  
+
 
 /*************************************************************************/
 /*! \brief Thread-safe implementation of strerror() */
 /**************************************************************************/
 char *gk_strerror(int errnum)
+/* comment this out as it is more trouble than it is worth! */
+/* { */
+/* #if defined(WIN32) || defined(__MINGW32__) */
```

Comment:
@nimgould 
The macro that is compiler independent on Windows is `_WIN32`.

Reply:
Understood. This part came from the metis people, and I suspect that they defined _WIN32 somewhere else. Unfortunately, the whole thing was out of date, and wouldn't work on macs, so I decided simply to go for the thread-unsafe version.